pkg/services/pluginsintegration/oauthserver/oauthserver.go

```diff
@@ -0,0 +1,59 @@
+package oauthserver
+
+import (
+	"context"
+
+	"github.com/grafana/grafana/pkg/plugins/oauth"
+	"github.com/grafana/grafana/pkg/services/oauthserver"
+)
+
+type Service struct {
+	os oauthserver.OAuth2Server
+}
+
+func ProvideService(os oauthserver.OAuth2Server) *Service {
+	s := &Service{
+		os: os,
+	}
+	return s
+}
+
+// SavePluginExternalService is a simplified wrapper around SaveExternalService for the plugin use case.
+func (s *Service) SavePluginExternalService(ctx context.Context, svcName string, svc *oauth.PluginExternalService) (*oauth.PluginExternalServiceRegistration, error) {
+	impersonation := oauthserver.ImpersonationCfg{
+		Permissions: svc.Impersonation.Permissions,
+	}
+	if svc.Impersonation.Enabled != nil {
```

Comment:
Possible nit/question. `svc.Impersonation` is not a pointer so cannot be nil - wouldn't this always enable impersonation even if not specified in the plugin.json? Should maybe `svc.Impersonation` be a pointer?

Similar question with `groups` if by default true - what would be the incentive to set that to false? My general feeling is enable as less as possible by default to limit the scope of permissions, but I'm not sure that's the correct way thinking about it?